tests/pathfinding/test_token_network.py

```diff
@@ -92,3 +92,30 @@ def test_graph_pruning():
     assert (
         len(pruned_p1_not_in_reachable_map.edges) == 2
     )  # just the two edges between 2 and 3 left
+
+
+def test_path_without_capacity(token_network_model: TokenNetwork, addresses: List[Address]):
+    """ Channels without capacity must not cause unexpected exceptions.
+
+    Regression test for https://github.com/raiden-network/raiden-services/issues/636
+    """
+    token_network_model.handle_channel_opened_event(
+        channel_identifier=ChannelID(1),
+        participant1=addresses[0],
+        participant2=addresses[1],
+        settle_timeout=15,
+    )
+    token_network_model.handle_channel_opened_event(
+        channel_identifier=ChannelID(2),
+        participant1=addresses[1],
+        participant2=addresses[2],
+        settle_timeout=15,
+    )
+
+    token_network_model.G[addresses[1]][addresses[2]]["view"].channel.capacity1 = 100
+    Path(
```

Comment:
Maybe we should also check that `Path.valid is True`? I'm scared that we simplify the constructor in the future this this becomes a useless test.

Reply:
The path is not valid. It can't receive a single token. But now it's invalid without throwing an exception.

Reply:
Yeah, then I mean a check that `Path.valid is False`. My point is that we check more than just running thew constructor, because we might simplify/change that soon.